File: src/com/mrpowergamerbr/loritta/commands/vanilla/music/VolumeCommand.java
package com.mrpowergamerbr.loritta.commands.vanilla.music;

import java.util.Arrays;
import java.util.List;

import com.mrpowergamerbr.loritta.LorittaLauncher;
import com.mrpowergamerbr.loritta.commands.CommandBase;
import com.mrpowergamerbr.loritta.commands.CommandCategory;
import com.mrpowergamerbr.loritta.commands.CommandContext;
import com.mrpowergamerbr.loritta.utils.music.GuildMusicManager;

import net.dv8tion.jda.core.Permission;

public class VolumeCommand extends CommandBase {
	@Override
	public String getLabel() {
		return "volume";
	}

	@Override
	public String getDescription() {
		return "Altera o volume da música";
	}

	@Override
	public List<String> getExample() {
		return Arrays.asList("5");
	}

	@Override
	public CommandCategory getCategory() {
		return CommandCategory.FUN;
	}

	@Override
	public void run(CommandContext context) {
		GuildMusicManager manager = LorittaLauncher.getInstance().getGuildAudioPlayer(context.getGuild());
		if (context.getHandle().hasPermission(Permission.MANAGE_SERVER)) {
			if (context.getArgs().length == 1) {	
				try {
					int vol = Integer.valueOf(context.getArgs()[0]);
					if (manager.player.getVolume() > vol) {
						context.sendMessage(context.getAsMention(true) + " irei diminuir o volume do batidão! Desculpe se eu te incomodei com a música alta...");
					} else {
						context.sendMessage(context.getAsMention(true) + " irei aumenta o volume do batidão! Se segura aí que agora você vai sentir as ondas sonoras!");
					}
					manager.player.setVolume(Integer.valueOf(context.getArgs()[0]));
				} catch (Exception e) {
					context.sendMessage(context.getAsMention(true) + " Ok, vamos alterar o volume para 💩 então... coloque um número válido por favor!");
				}
			} else {
				this.explain(context.getConfig(), context.getEvent());
			}
		}
	}
}

File: src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/music/VolumeCommand.kt
package com.mrpowergamerbr.loritta.commands.vanilla.music

import com.mrpowergamerbr.loritta.commands.AbstractCommand
import com.mrpowergamerbr.loritta.commands.CommandContext
import com.mrpowergamerbr.loritta.network.Databases
import com.mrpowergamerbr.loritta.utils.Constants
import com.mrpowergamerbr.loritta.utils.LoriReply
import com.mrpowergamerbr.loritta.utils.locale.LegacyBaseLocale
import com.mrpowergamerbr.loritta.utils.loritta
import com.mrpowergamerbr.loritta.utils.save
import net.dv8tion.jda.api.Permission
import net.perfectdreams.loritta.api.commands.CommandCategory
import org.jetbrains.exposed.sql.transactions.transaction
import java.util.*

class VolumeCommand : AbstractCommand("volume", category = CommandCategory.MUSIC) {
	override fun getDescription(locale: LegacyBaseLocale): String {
		return locale.get("VOLUME_DESCRIPTION")
	}

	override fun getExamples(): List<String> {
		return Arrays.asList("100", "66", "33")
	}

	override fun getDiscordPermissions(): List<Permission> {
		return listOf(Permission.VOICE_MUTE_OTHERS)
	}

	override fun requiresMusicEnabled(): Boolean {
		return true
	}

	override fun canUseInPrivateChannel(): Boolean {
		return false
	}

	override suspend fun run(context: CommandContext,locale: LegacyBaseLocale) {
		val serverConfig = loritta.getOrCreateServerConfig(context.guild.idLong)

		val donationKey = transaction(Databases.loritta) {
			serverConfig.donationKey
		}

		val donatedMoney = loritta.getActiveMoneyFromDonations(context.userHandle.idLong)

		if ((donationKey == null || !donationKey.isActive() || 19.99 > donationKey.value) && 19.99 > donatedMoney) {
			context.reply(
					locale["PREMIUM_CantUseFeature"],
					"\uD83D\uDCB8"
			)
			return
		}

		val manager = loritta.audioManager.getGuildAudioPlayer(context.guild)
		if (context.args.isNotEmpty()) {
			try {
				val vol = Integer.valueOf(context.args[0])
				if (vol > 500) {
					context.sendMessage(Constants.ERROR + " **|** " + context.getAsMention(true) + locale["VOLUME_TOOHIGH"])
					return
				}
				if (0 > vol) {
					context.sendMessage(Constants.ERROR + " **|** " + context.getAsMention(true) + locale["VOLUME_TOOLOW"])
					return
				}

				manager.player.volume = context.args[0].toInt()

				// context.config.volume = manager.player.volume

				loritta save context.config

				if (manager.player.volume > vol) {
					context.reply(
							LoriReply(
									message = locale["VOLUME_LOWER"],
									prefix = "\uD83D\uDD08"
							)
					)
				} else {
					context.reply(
							LoriReply(
									message = locale["VOLUME_HIGHER"],
									prefix = "\uD83D\uDD0A"
							)
					)
				}
			} catch (e: Exception) {
				context.sendMessage(Constants.ERROR + " **|** " + context.getAsMention(true) + locale["VOLUME_EXCEPTION"])
			}
		} else {
			context.explain()
		}
	}
}